### PR TITLE
Rework template assets to use variables instead of component specific param lists

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: flows/actions/testdata/_assets.json [hunk 1]
```diff
@@ -333,34 +342,35 @@
                             "type": "header",
                             "name": "header",
                             "content": "",
-                            "params": [
-                                {
-                                    "type": "image"
-                                }
-                            ]
```
Comment (rowanseymour) [inline flows/actions/testdata/_assets.json:340]: stripping this out because we don't support anything besides text

File: flows/actions/send_msg.go [hunk 2]
```diff
@@ -161,15 +162,19 @@ func (a *SendMsgAction) getTemplateMsg(run flows.Run, urn urns.URN, channelRef *
...
 		paramValues := evaluatedParams[comp.Name()]
-		params := make([]flows.TemplatingParam, len(comp.Params()))
+		params := make([]flows.TemplatingParam, len(varMap))
 
-		for i, p := range comp.Params() {
+		for i, varName := range utils.SortedKeys(varMap) {
+			v := variables[varMap[varName]]
 			if i < len(paramValues) {
-				params[i] = flows.TemplatingParam{Type: p.Type(), Value: paramValues[i]}
+				params[i] = flows.TemplatingParam{Type: v.Type(), Value: paramValues[i]}
```
Comment (norkans7) [inline flows/actions/send_msg.go:175]: Any reason to not add `varName` in the `TemplatingParam`?
That is the new data we need to be able to support the Twilio templates which triggered this refactoring.
I guess that could be another PR too if this is just for backward compatibility
